liquigraph-core/src/main/java/org/liquigraph/core/io/ChangelogGraphReader.java

```diff
@@ -16,89 +16,14 @@
-            "WHERE EXISTS(exec.`order`) AND EXISTS(changeset.query) " +
-            "SET exec.time = exec.`order` " +
-            "WITH exec, changeset " +
-            "REMOVE exec.`order` " +
-            "WITH changeset " +
-            "CREATE (changeset)<-[:EXECUTED_WITHIN_CHANGESET {`order`: 0}]-(query:__LiquigraphQuery) " +
-            "SET query.query = changeset.query " +
-            "WITH changeset " +
-            "REMOVE changeset.query";
-
-    private static final String MATCH_CHANGESETS =
-        "MATCH (:__LiquigraphChangelog)<-[changelog_execution:EXECUTED_WITHIN_CHANGELOG]-(changeset:__LiquigraphChangeset), " +
-        "      (changeset)<-[changeset_execution:EXECUTED_WITHIN_CHANGESET]-(query:__LiquigraphQuery) " +
-        "WITH changelog_execution.`time` AS changeset_position, changeset, query " +
-        "ORDER BY changeset_position ASC, changeset_execution.`order` ASC " +
-        "WITH changeset_position, changeset, COLLECT(query.query) AS queries " +
-        "RETURN {" +
-        "   id: changeset.id, " +
-        "   author:changeset.author, " +
-        "   checksum:changeset.checksum, " +
-        "   query:queries" +
-        "} AS changeset " +
-        "ORDER BY changeset_position ASC";
-
-    private static final Logger LOGGER = LoggerFactory.getLogger(ChangelogFileWriter.class);
-
-    public final Collection<Changeset> read(Connection connection) {
-        Collection<Changeset> changesets = new ArrayList<>();
-        try (Statement statement = connection.createStatement()) {
-            LOGGER.debug("Migrating pre 1.0-RC3 history graph");
-            statement.execute(MIGRATE_PRE_1_0_RC3_CHANGELOG);
-            try (ResultSet result = statement.executeQuery(MATCH_CHANGESETS)) {
-                while (result.next()) {
-                    changesets.add(mapRow(result.getObject("changeset")));
-                }
-            }
-            connection.commit();
-            LOGGER.debug("Retrieved {} changesets", changesets.size());
-            return changesets;
-        } catch (SQLException e) {
-            throw propagate(e);
-        }
-    }
-
-    @SuppressWarnings("unchecked")
-    private Changeset mapRow(Object line) throws SQLException {
-        if (line instanceof Map) {
-            return changeset((Map<String, Object>) line);
-        }
-        throw new IllegalArgumentException(format(
-           "Unsupported row.\n\t" +
-           "Cannot parse: %s", line));
-    }
-
-    private Changeset changeset(Map<String, Object> node) {
-        Changeset changeset = new Changeset();
-        changeset.setAuthor(String.valueOf(node.get("author")));
-        changeset.setId(String.valueOf(node.get("id")));
-        changeset.setQueries(adaptQueries(node.get("query")));
-        changeset.setChecksum(String.valueOf(node.get("checksum")));
-        return changeset;
-    }
+/**
+ * Interface to retrieve the persisted changelog from neo4j db.
+ */
+public interface ChangelogGraphReader {
```

Comment:
per our previous discussion (about implementing clearChecksums as a private method in ChangelogGraphReader, this change won't be necessary

Reply:
done in 167e6deda602faa5e88b77fc4d14e246a8b4fcf5